MDNetworking/MemberImpl/MDReplicatedMember.cs

```diff
@@ -0,0 +1,165 @@
+using Godot;
+using System;
+using System.Reflection;
+
+public class MDReplicatedMember
+{
+    public bool ProcessWhilePaused { get; set; } = true;
+
+    public string ReplicationGroup { get; set; } = null;
+
+    protected const string LOG_CAT = "LogReplicatedMember";
+
+    protected MemberInfo Member;
+
+    protected object LastValue;
+
+    protected bool Reliable;
+
+    protected MDReplicatedType ReplicatedType;
+
+    protected WeakRef NodeRef;
+
+    protected bool IsShouldReplicate = false;
+
+    public MDReplicatedMember(MemberInfo Member, bool Reliable, MDReplicatedType ReplicatedType, WeakRef NodeRef, MDReplicatedSetting[] Settings)
+    {
+        MDLog.AddLogCategoryProperties(LOG_CAT, new MDLogProperties(MDLogLevel.Info));
+        this.Member = Member;
+        this.Reliable = Reliable;
+        this.ReplicatedType = ReplicatedType;
+        this.NodeRef = NodeRef;
+        CheckIfShouldReplicate();
+    }
+
+    ///<summary>Proviedes a unique key to identify this member</summary>
+    public String GetUniqueKey()
+    {
+        Node Node = NodeRef.GetRef() as Node;
+        return Node.GetPath() + "#" + Member.Name;
+    }
+
+    protected virtual object GetValue()
+    {
+        Node Instance = NodeRef.GetRef() as Node;
+        FieldInfo Field = Member as FieldInfo;
+
+        if (Field != null)
+        {
+            return Field.GetValue(Instance);
+        }
+
+        PropertyInfo Property = Member as PropertyInfo;
+        if (Property != null)
+        {
+            return Property.GetValue(Instance);
+        }
+
+        return null;
+    }
+
+    ///<summary>It is better to implement ReplicateToAll and ReplicateToPeer instead of this method</summary>
+    public virtual void Replicate(int JoinInProgressPeerId, bool IsIntervalReplicationTime)
+    {
+        object CurrentValue = GetValue();
+        Node Instance = NodeRef.GetRef() as Node;
+
+        if ((ReplicatedType == MDReplicatedType.Interval && IsIntervalReplicationTime) || 
+        (object.Equals(LastValue, CurrentValue) == false && ReplicatedType == MDReplicatedType.OnChange))
+        {
+            ReplicateToAll(Instance, CurrentValue);
+        }
+        else if (JoinInProgressPeerId != -1)
+        {
+            ReplicateToPeer(Instance, CurrentValue, JoinInProgressPeerId);
+        }
+    }
+
+    ///<summary>Replicate this value to all clients</summary>
+    protected virtual void ReplicateToAll(Node Node, object Value)
+    {
+        MDLog.Debug(LOG_CAT, "Replicating {0} with value {1} from {2}", Member.Name, Value, LastValue);
+        GD.Print("Replicating " + Member.Name + " from node " + Node.Name + " with value " + Value);
```

Comment:
Maybe we should add GD.Print to the Log calls.

Reply:
This was something left over from my own testing code, I will remove this. But if we can hook into GD.Print that would be great